src/main/java/org/llorllale/cactoos/matchers/IsNumber.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) for portions of project cactoos-matchers are held by
+ * Yegor Bugayenko, 2017-2018, as part of project cactoos.
+ * All other copyright for project cactoos-matchers are held by
+ * George Aristy, 2018-2020.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.llorllale.cactoos.matchers;
+
+import java.util.Comparator;
+
+/**
+ * Matcher for {@link Number}.
+ *
+ * @since 1.0.0
+ */
+public final class IsNumber extends MatcherEnvelope<Number> {
```

Comment:
@andreoss should this be called `IsNumber` or instead `NumberHasValue`? because I would expect `IsNumber` to check that an object is a number in general…

Reply:
@victornoel Hamcrest itself has `IsNaN` and `IsCloseTo`. I believe `Has` and `Is` prefixes on matchers should be about "is-a"/"has-a" relationships.  Also there's `TextIs`

Reply:
@andreoss Well, `IsNaN` is ok because it doesn't take an argument, that was my point. `TextIs` is relevant since you can read it naturally: `TextIs("bla")`. So `NumberIs` is better than `IsNumber` for our usage then, let's go with that?